sharc/results.py

```diff
@@ -23,449 +24,89 @@ def __init__(self, parameters_filename: str,
         
-        self.imt_ul_tx_power_density = list()
-        self.imt_ul_tx_power = list()
-        self.imt_ul_sinr_ext = list()
-        self.imt_ul_sinr = list()
-        self.imt_ul_snr = list()
-        self.imt_ul_inr = list()
-        self.imt_ul_tput_ext = list()
-        self.imt_ul_tput = list()
-
-        self.imt_path_loss = list()
-        self.imt_coupling_loss = list()
-        self.imt_bs_antenna_gain = list()
-        self.imt_ue_antenna_gain = list()
-
-        self.system_imt_antenna_gain = list()
-        self.imt_system_antenna_gain = list()
-        self.imt_system_path_loss = list()
-        self.imt_system_build_entry_loss = list()
-        self.imt_system_diffraction_loss = list()
-
-        self.imt_dl_tx_power_density = list()
-        self.imt_dl_tx_power = list()
-        self.imt_dl_sinr_ext = list()
-        self.imt_dl_sinr = list()
-        self.imt_dl_snr = list()
-        self.imt_dl_inr = list()
-        self.imt_dl_tput_ext = list()
-        self.imt_dl_tput = list()
-
-        self.system_ul_coupling_loss = list()
-        self.system_ul_interf_power = list()
-
-        self.system_dl_coupling_loss = list()
-        self.system_dl_interf_power = list()
-
-        self.system_inr = list()
-        self.system_pfd = list()
-        self.system_rx_interf = list()
-
-        self.plot_list = None
-
+        self.data = {
+            "imt_ul_tx_power_density": [],
+            "imt_ul_tx_power": [],
+            "imt_ul_sinr_ext": [],
+            "imt_ul_sinr": [],
+            "imt_ul_snr": [],
+            "imt_ul_inr": [],
+            "imt_ul_tput_ext": [],
+            "imt_ul_tput": [],
+            "imt_path_loss": [],
+            "imt_coupling_loss": [],
+            "imt_bs_antenna_gain": [],
+            "imt_ue_antenna_gain": [],
+            "system_imt_antenna_gain": [],
+            "imt_system_antenna_gain": [],
+            "imt_system_path_loss": [],
+            "imt_system_build_entry_loss": [],
+            "imt_system_diffraction_loss": [],
+            "imt_dl_tx_power_density": [],
+            "imt_dl_tx_power": [],
+            "imt_dl_sinr_ext": [],
+            "imt_dl_sinr": [],
+            "imt_dl_snr": [],
+            "imt_dl_inr": [],
+            "imt_dl_tput_ext": [],
+            "imt_dl_tput": [],
+            "system_ul_coupling_loss": [],
+            "system_ul_interf_power": [],
+            "system_dl_coupling_loss": [],
+            "system_dl_interf_power": [],
+            "system_inr": [],
+            "system_pfd": [],
+            "system_rx_interf": []
+        }
+
+        self.statistics = {} # To store the computed statistics
         self.__sharc_dir = pathlib.Path(__file__).parent.resolve()
         self.output_dir_parent = output_dir
-
         if not overwrite_output:
             today = datetime.date.today()
-
             results_number = 1
-            results_dir_head = output_dir_prefix + '_' + today.isoformat() + '_' + "{:02n}"
```

Comment:
Essas implementações não são equivalentes. Antes ele chamava recursivamente o create_dir até encontrar um `results_number` que dá pra criar o diretório (ou seja, que o `results_number` ainda não foi usado)

Agora ele sempre sobrescreve o número 1.

Se isso for o comportamento desejado, a flag verificada em `if not overwrite_output:` não está fazendo sentido

Reply:
Obrigado pelo comentario @artistrea, acabei de subir um commit que tenta resolver o que voce apontou. Pode dar uma olhada se esta de acordo com o que voce disse?